### PR TITLE
feat(job): validate UnrecoverableError presence

### PR DESCRIPTION
**Pull request by roggervalf**

### CODE REVIEW

File: src/classes/unrecoverable-error.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/**
+ * UnrecoverableError
+ *
+ * Error to pass a job to failed even if the attemptsMade
```
Comment (manast) [inline src/classes/unrecoverable-error.ts:4]: pass -> move

File: src/classes/job.ts [hunk 2]
```diff
@@ -442,8 +443,8 @@ export class Job<
    * @param fetchNext - true when wanting to fetch the next job
    * @returns void
    */
-  async moveToFailed(
-    err: Error,
+  async moveToFailed<E extends Error>(
```
Comment (manast) [inline src/classes/job.ts:446]: Will this work for "sandboxed" processors?

File: src/classes/unrecoverable-error.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+ * are lower than the expected limit.
+ *
+ */
+export class UnrecoverableError extends Error {
+  constructor(message?: string) {
+    super(message);
+    this.name = this.constructor.name;
+    Object.setPrototypeOf(this, new.target.prototype);
```
Comment (roggervalf) [inline src/classes/unrecoverable-error.ts:12]: base on https://stackoverflow.com/questions/41102060/typescript-extending-error-class

File: src/classes/job.ts (hunk 2 above)
Comment (roggervalf) [reply src/classes/job.ts:446]: yes, it will work, I also added a test case for this

**Review by manast — APPROVED** [review]
Good job!

**Pull request merged**